doc/developers/release_process.rst

```diff
@@ -0,0 +1,311 @@
+Release Process
+===============
+
+This section is about preparing a major release, incrementing the minor
+version, or a bug fix release incrementing the patch version. Our convention is
+that we release one or more release candidates (0.RRrcN) before releasing the
+final distributions. We follow the `PEP101
+<https://www.python.org/dev/peps/pep-0101/>`_ to indicate release candidates,
+post, and minor releases.
+
+Before a release
+----------------
+
+1. Update authors table:
+
+   .. prompt:: bash $
+
+       cd build_tools; make authors; cd ..
+
+   and commit. This is only needed if the authors have changed since the last
+   release. This step is sometimes done independent of the release. This
+   updates the maintainer list and is not the contributor list for the release.
+
+2. Confirm any blockers tagged for the milestone are resolved, and that other
+   issues tagged for the milestone can be postponed.
+
+3. Ensure the change log and commits correspond (within reason!), and that the
+   change log is reasonably well curated. Some tools for these tasks include:
+
+   - ``maint_tools/sort_whats_new.py`` can put what's new entries into
+     sections. It's not perfect, and requires manual checking of the changes.
+     If the what's new list is well curated, it may not be necessary.
+
+   - The ``maint_tools/whats_missing.sh`` script may be used to identify pull
+     requests that were merged but likely missing from What's New.
+
+4. Make sure the deprecations, FIXME and TODOs tagged for the release have
+   been taken care of.
+
+**Permissions**
+
+The release manager requires a set of permissions on top of the usual
+permissions given to maintainers, which includes:
+
+- *maintainer* role on ``scikit-learn`` projects on ``pypi.org`` and
+  ``test.pypi.org``, separately.
+- become a member of the *scikit-learn* team on conda-forge by editing the
+  ``recipe/meta.yaml`` file on
+  ``https://github.com/conda-forge/scikit-learn-feedstock``
```

Comment:
The PyPI part is not strict requirement anymore, We should say that all you need is to be a core dev to be able to upload a release on PyPI, and that being admin on PyPI allows to manage a release in case something went wrong.

Reply:
Let me know if 5944d984226518f621bf85955e4f052221a368ae addresses your problem.